File: dev_page/src/main/java/org/example/page/task/signin/Service.java
package org.example.page.task.signin;

import org.example.core.page.application.AbstractService;
import org.example.core.page.cookie.ICookie;
import org.example.page.cookie.store.Role;

class Service extends AbstractService {
    private final Request request;

    Service(Request request, ICookie cookie) {
        super(cookie);
        this.request = request;
    }

    void login(String id, String password) {
        request.login(new EntityUser(id, password));
        if (cookie instanceof Role role) {
            role.setId(id);
            role.setRole("temporary");
        }
    }
}

File: dev-page/src/main/kotlin/org/example/page/task/signin/Service.kt
package org.example.page.task.signin

import org.example.core.page.application.AbstractService
import org.example.core.page.cookie.ICookie
import org.example.page.cookie.store.Role

internal class Service(private val request: Request, cookie: ICookie?) : AbstractService(cookie) {
    fun login(id: String?, password: String?) {
        request.login(EntityUser(id, password))

        // cookie를 Role로 안전하게 형변환한 뒤, 성공(null이 아님)하면 실행
        (cookie as? Role)?.apply {
            this.id = id
            this.role = "temporary"
        }
    }
}